### ISSUE TITLE
Broken CI, Update deps

### ISSUE THREAD

**Issue by estk** [alecmocatta/palaver#47]
It looks like CI is not running correctly anymore and therefore the dependabot PR's are not passing. Would you be interested in a PR changing to GH ci?

**Comment by alecmocatta** [comment]
The CI is failing because it's linting with a `rustc` version < the MSRV of the `bitflags` crate, as far as I can tell. I'm bumping it here #48 which should fix CI.

I would love however if it could be switched to GitHub Actions. The only requirement is that I'd like to to test most or all of the permutations that are currently tested here against stable and nightly:

https://github.com/alecmocatta/palaver/blob/60f5e88882fc7f0e4f1427cce6cef57ac291ee88/azure-pipelines.yml#L24-L35

**Comment by alecmocatta** [comment]
\* Correction. CI passes tests on mac and windows, but fails running them under `miri`. CI fails to install `miri` on linux.

To fix current CI:
* `#[cfg_attr(miri, ignore)]` tests that fail under `miri`
* Bump the CI rustc lint version, which should fix CI on Linux

**Comment by estk** [comment]
Looks great, anything i can lend a hand with for deps update?

**Comment by alecmocatta** [comment]
Absolutely, a PR that updates all deps would be much appreciated. On Azure pipelines as long as it passes "Run tests" (under "test windows", "test mac" and "test linux") then I'm very happy to merge.
